platform/openide.nodes/src/org/openide/nodes/PropertySupport.java

```diff
@@ -360,6 +522,35 @@ public ReadWrite(String name, Class<T> type, String displayName, String shortDes
         }
     }
 
+    private static final class FunctionalProperty<T> extends PropertySupport<T> {
+        private final Supplier<T> supplier;
+        private final Consumer<T> consumer;
+
+        public FunctionalProperty(String name, Class<T> type, String displayName, String shortDescription, Supplier<T> supplier, Consumer<T> consumer) {
+            super(name, type, displayName, shortDescription, supplier != null, consumer != null);
+            this.supplier = supplier;
+            this.consumer = consumer;
+        }
+
+        @Override
+        public T getValue() throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```

Comment:
I guess there's no need to declare the exceptions, probably a leftover from a reflective op copypaste.

Reply:
Yes, copypaste indeed. Good catch!